src/tribler-gui/tribler_gui/widgets/settingspage.py

```diff
@@ -223,6 +233,78 @@ def initialize_with_settings(self, settings):
         connect(self.window().slider_cpu_level.valueChanged, self.show_updated_cpu_priority)
         self.window().checkbox_enable_network_statistics.setChecked(settings['ipv8']['statistics'])
 
+    def _version_dir_checkbox(self, version_dir, enabled=True):
+        text = f"{format_size(get_dir_size(version_dir))}   {version_dir}"
+        checkbox = QCheckBox(text)
+        checkbox.setEnabled(enabled)
+        return checkbox
+
+    def load_settings_data_tab(self):
+        self.refresh_current_version_checkbox()
+        self.refresh_old_version_checkboxes()
+
+    def refresh_current_version_checkbox(self):
+        root_state_dir = get_root_state_directory()
+        current_version_dir = get_versioned_state_directory(root_state_dir)
+        self.refresh_version_checkboxes(self.window().state_dir_current, [current_version_dir], enabled=False)
+
+    def refresh_old_version_checkboxes(self):
+        root_state_dir = get_root_state_directory()
+        old_version_dirs = get_installed_versions(root_state_dir, current_version=False)
+        self.refresh_version_checkboxes(self.window().state_dir_list, old_version_dirs, enabled=True)
+
+    def refresh_version_checkboxes(self, parent, versions, enabled=True):
+        version_dirs_layout = parent.layout()
+        for checkbox in parent.findChildren(QCheckBox):
+            version_dirs_layout.removeWidget(checkbox)
+            checkbox.setParent(None)
+            checkbox.deleteLater()
+
+        for old_version_dir in versions:
+            version_dir_checkbox = self._version_dir_checkbox(old_version_dir, enabled=enabled)
+            version_dirs_layout.addWidget(version_dir_checkbox)
+
+    def on_remove_version_dirs(self, _):
+        root_version_dir = str(get_root_state_directory())
+
+        def version_from_checkbox_text(checkbox):
+            # eg text: 5 GB /home/<user>/.Tribler/v7.8
+            return checkbox.text().split("/")[-1].replace(root_version_dir, "")
+
+        versions_selected_for_deletion = []
+        for checkbox in self.window().state_dir_list.findChildren(QCheckBox):
+            if checkbox.isChecked():
+                versions_selected_for_deletion.append(version_from_checkbox_text(checkbox))
+
+        if self.on_confirm_remove_version_dirs(versions_selected_for_deletion):
+            remove_version_dirs(root_version_dir, versions_selected_for_deletion)
+            self.refresh_old_version_checkboxes()
+
+    def on_confirm_remove_version_dirs(self, selected_versions):
+        message_box = QMessageBox()
+        message_box.setIcon(QMessageBox.Question)
+
+        if selected_versions:
+            version_dirs_str = "\n- ".join(selected_versions)
+            versions_info = f"Versions: \n- {version_dirs_str}"
+
+            title = "Confirm delete older versions?"
+            message_body = f"Are you sure to remove the selected versions? " \
```

Comment:
Better to
"Are you sure you want to remove the state directories for selected versions?"